default/python/freeOrionAIInterface.py

```diff
@@ -1,3 +1,6 @@
+# Autogenerated do not modify manually!
```

Comment:
If we left this as a 'mock' freeOrionAIInterface.py, this comment would need to be augmented with an explanation of what the file is.   But it's one thing to have a mock file tucked away into a directory labeled as such, it's another to move it into one of our regular directories.

What I really I think we need to do is make this into a [Pep 484 stub file](https://www.python.org/dev/peps/pep-0484/#stub-files), which don't seem to often carry such comments, though I think it would still be helpful to lead with some lines like
```
# Autogenerated do not modify manually!
# This is a type-hinting python stub file, used by python IDEs to provide type hints.  For more information
# about stub files, see https://www.python.org/dev/peps/pep-0484/#stub-files
# During execution, the actual freeOrionAIInterface module is made available to the AI via
# a C++ Boost-python process as part of the AI launch.
```

Although the current form of the file has a lot of statements that don't seem to me like they would normally go into a stub file, a review of the PEP indicates that they don't seem to be barred, either, and in my brief tests they did not seem to cause any trouble.  Rather than moving this file to ```default/python/freeOrionAIInterface.py```, we can simply rename and move it to ```default/python/AI/freeOrionAIInterface.pyi``` and ouila! We have a working stub file.

Changes to the format/contents of the file, to make it more in line with standard stub files, could come later, it seems to me.

@Morlic-fo @LGM-Doyle Do either of you have much experience with python stub files yet?  (I'm not sure which IDEs beyond Pycharm and Wing actually support them yet.)  I invite discussion about the approach I suggest here.